sycl/include/sycl/multi_ptr.hpp

```diff
@@ -113,6 +113,16 @@ class multi_ptr {
       : m_Pointer(ptr) {}
   multi_ptr(std::nullptr_t) : m_Pointer(nullptr) {}
 
+  // Implicit conversion from multi_ptr<T> to multi_ptr<const T>
+  template <
+      typename _ElementType,
+      typename = typename std::enable_if_t<
+          std::is_const_v<ElementType> && !std::is_const_v<_ElementType> &&
+          std::is_same_v<std::remove_const_t<ElementType>, _ElementType>>>
+  explicit multi_ptr(typename multi_ptr<_ElementType, Space,
+                                        access::decorated::yes>::pointer ptr)
+      : multi_ptr<ElementType, Space, access::decorated::yes>(ptr) {}
```

Comment:
Just a quick late question: Shouldn't this be

```suggestion
  template <
      typename NonConstElementType = std::remove_const_t<ElementType>,
      typename = typename std::enable_if_t<
          std::is_const_v<ElementType> &&
          std::is_same_v<NonConstElementType, std::remove_const_t<ElementType>>>>
  explicit multi_ptr(typename multi_ptr<NonConstElementType, Space,
                                        access::decorated::yes>::pointer ptr)
      : multi_ptr(ptr) {}
```
otherwise I don't see how we would ever be able to deduce `_ElementType`, seeing as `::pointer` is an alias. Is this ctor actually in use anywhere?

Reply:
Since these interfaces are old and not part of SYCL 2020, I have opened a separate PR with the SYCL 2020 equivalents (deprecating the old ones): https://github.com/intel/llvm/pull/9377. Once that is in, I don't think we need to change the `async_work_group_copy` interfaces in this PR.

Reply:
> 
I had a test for that ctor. The `_ElementType ` is part of template arguments of the ctor.
[c'tor test]([sycl/test/multi_ptr/ctad.cpp](https://github.com/intel/llvm/pull/8874/commits/eae8bcbd89f4b540f303d600bb78dbd03b130929#diff-879c9a07353f3f9f7e34e89b2457f3310d24908e61706618c9efce3e73decc41))

Happy to change if you see any issue with it.

Reply:
Thanks @steffenlarsen. You are right. I updated the implicit conversion.